src/runtime/opencl.cpp

```diff
@@ -130,6 +134,24 @@ WEAK const char *halide_opencl_get_device_type(void *user_context) {
     return device_type;
 }
 
+WEAK void halide_opencl_set_build_options(const char *n) {
+    if (n) {
```

Comment:
Why don't we need to acquire build_options_lock here?

Reply:
I think we do, added here and in other two existing functions.